### PR TITLE
Metadata backwards compatible and more performant Closes #936

### LINKED ISSUE #936: Metadata REST API should be backwards compatible
Currently the new metadata API is on V2, it should be V1 and be made to have the same format as V1 (ie no V2)

One specific change is that the current implementation (V2) returns a map of workflow id -> metadata.  This should be flattened (no workflow id) because it is a single workflow REST API.

If there are other differences between the current and former REST api metadata, let's quickly techtalk those differences and the PO will decide

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

**Comment by Horneth** [comment]
Sorry for the double-penalty @mcovarr @cjllanwarne 
The changes to make the Metadata backwards compatible are exactly the same, I just re-arranged the algorithm to make it pretty much 100x faster. I can explain how if people are interested

File: engine/src/main/scala/cromwell/webservice/metadata/IndexedJsonValue.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          case (o1: TimestampedJsList, o2: TimestampedJsList) =>
+            val sg = implicitly[Semigroup[Map[Int, TimestampedJsValue]]]
+            TimestampedJsList(sg.append(o1.v, o2.v), dateTimeOrdering.max(o1.timestamp, o2.timestamp))
+          case (o1, o2) => timestampedJsValueOrdering.max(o1, o2)
+        }
+      }
+
+      override def zero: TimestampedJsValue = TimestampedJsObject(Map.empty, OffsetDateTime.now)
+    }
+  }
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/webservice/metadata/IndexedJsonValue.scala]: superfluous level of curlies

File: engine/src/main/scala/cromwell/webservice/metadata/MetadataBuilderActor.scala [hunk 2]
```diff
@@ -93,17 +63,22 @@ object MetadataBuilderActor {
...
-          IndexedJsonList(Vector(acc.withIndex(index)))
-        })
+        val list = innerValue match {
+           // Empty value in a list means empty list
+          case TimestampedEmptyJson(timestamp) => TimestampedJsList(Map.empty, timestamp)
+          case nonEmptyValue =>
+            val innerList = TimestampedJsList(TreeMap(indexes.last.toInt -> innerValue), nonEmptyValue.timestamp)
+            indexes.init.foldRight(innerList)((index, acc) => {
+              TimestampedJsList(TreeMap(index.toInt -> acc), acc.timestamp)
+            })
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/webservice/metadata/MetadataBuilderActor.scala]: superfluous curlies

Comment (mcovarr) [inline engine/src/main/scala/cromwell/webservice/metadata/MetadataBuilderActor.scala]: could you please drop a comment as to what this is doing?  I can't figure it out.  😦

**Comment by cjllanwarne** [comment]
Lots of stuff that's gone over my head, but... I can't see any coding issues 👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/cromwell/pull-request/981/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/cromwell)

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**